src/main/java/hudson/tasks/MailSender.java

```diff
@@ -177,26 +177,31 @@ protected MimeMessage getMail(AbstractBuild<?, ?> build, BuildListener listener)
         } catch (NoSuchMethodException x) {
             // non-deprecated subclass
         }
+
         if (build.getResult() == Result.FAILURE) {
             return createFailureMail(build, listener);
         }
-
         if (build.getResult() == Result.UNSTABLE) {
-            if (!dontNotifyEveryUnstableBuild)
+            if (!dontNotifyEveryUnstableBuild) {
                 return createUnstableMail(build, listener);
+            }
             Result prev = findPreviousBuildResult(build);
-            if (prev == Result.SUCCESS || prev == null)
+            if (prev == Result.SUCCESS || prev == null) {
                 return createUnstableMail(build, listener);
+            }
         }
 
         if (build.getResult() == Result.SUCCESS) {
             Result prev = findPreviousBuildResult(build);
-            if (prev == Result.FAILURE)
+            if (prev == Result.FAILURE) {
                 return createBackToNormalMail(build, Messages.MailSender_BackToNormal_Normal(), listener);
-            if (prev == Result.UNSTABLE)
+            }
+            if (prev == Result.UNSTABLE) {
                 return createBackToNormalMail(build, Messages.MailSender_BackToNormal_Stable(), listener);
+            }
         }
 
+        System.out.println("Not sending any mail as BuildResult is not set at all.");
```

Comment:
You should use a logger here.

Reply:
What is the "standard" way for logging in Jenkins plugins? Log4j? Slf4J? Therefore I thought that StandardOut-Logging would result in an output in the job console .....

Reply:
The standard way is either directly through java.util.logging or via SLF4J.

Reply:
I'm not sure if java.util.logging works, but replaced the line in the code - thanks.